--- confluent-5.2.1/bin/kafka-run-class.~1~	2019-04-02 07:09:20.000000000 +0000
+++ confluent-5.2.1/bin/kafka-run-class	2019-05-22 19:12:09.667029790 +0000
@@ -14,6 +14,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if [ -f /etc/kafka/env ]; then
+	. /etc/kafka/env
+fi
+
 if [ $# -lt 1 ];
 then
   echo "USAGE: $0 [-daemon] [-name servicename] [-loggc] classname [opts]"
--- confluent-5.2.1/bin/kafka-rest-run-class.~1~        2019-05-23 10:38:01.857541388 +0000
+++ confluent-5.2.1/bin/kafka-rest-run-class    2019-05-23 10:38:25.396946348 +0000
@@ -14,6 +14,10 @@
 # specific language governing permissions and limitations under the License.
 #

+if [ -f /etc/kafka-rest/env ]; then
+  . /etc/kafka-rest/env
+fi
+
 base_dir=$(dirname $0)/..

 # Development jars. `mvn package` should collect all the required dependency jars here
--- confluent-5.2.1/bin/schema-registry-run-class.~1~   2019-05-23 10:40:21.464570836 +0000
+++ confluent-5.2.1/bin/schema-registry-run-class       2019-05-23 10:40:38.677152767 +0000
@@ -15,6 +15,10 @@
 # limitations under the License.
 #

+if [ -f /etc/schema-registry/env ]; then
+  . /etc/schema-registry/env
+fi
+
 if [ $# -lt 1 ];
 then
   echo "USAGE: $0 [-daemon] [-name servicename] [-loggc] classname [opts]"
--- confluent-5.2.1/bin/ksql-run-class.~1~      2019-05-23 10:42:11.705388907 +0000
+++ confluent-5.2.1/bin/ksql-run-class  2019-05-23 10:42:28.597062827 +0000
@@ -3,6 +3,10 @@

 # Use shellcheck to lint this file

+if [ -f /etc/ksql/env ]; then
+ . /etc/ksql/env
+fi
+
 base_dir=$( cd -P "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )

 # Log directory to use
