--- graylog-2.4.6/bin/graylogctl.orig	2018-11-27 18:51:54.728067938 +0000
+++ graylog-2.4.6/bin/graylogctl	2018-11-27 18:53:22.014951324 +0000
@@ -52,6 +52,7 @@
 LOG_FILE=${LOG_FILE:=log/graylog-server.log}
 LOG4J=${LOG4J:=}
 DEFAULT_JAVA_OPTS="-Djava.library.path=${GRAYLOGCTL_DIR}/../lib/sigar -Xms1g -Xmx1g -XX:NewRatio=1 -server -XX:+ResizeTLAB -XX:+UseConcMarkSweepGC -XX:+CMSConcurrentMTEnabled -XX:+CMSClassUnloadingEnabled -XX:+UseParNewGC -XX:-OmitStackTraceInFastThrow"
+DEFAULT_JAVA_OPTS="$DEFAULT_JAVA_OPTS -XX:-UseParNewGC -XX:-UseConcMarkSweepGC -XX:+UseCondCardMark -XX:MaxGCPauseMillis=200 -XX:+UseG1GC -XX:GCPauseIntervalMillis=1000 -XX:InitiatingHeapOccupancyPercent=35"
 JAVA_OPTS="${JAVA_OPTS:="$DEFAULT_JAVA_OPTS"}"
 
 start() {
