--- graylog-3.1.0/bin/graylogctl.~1~	2019-08-27 09:13:50.427481701 +0000
+++ graylog-3.1.0/bin/graylogctl	2019-08-27 09:17:18.027555196 +0000
@@ -55,7 +55,7 @@
 if $JAVA_CMD -XX:+PrintFlagsFinal 2>&1 |grep -q UseParNewGC; then
 	DEFAULT_JAVA_OPTS="${DEFAULT_JAVA_OPTS} -XX:+UseParNewGC"
 fi
-JAVA_OPTS="${JAVA_OPTS:="$DEFAULT_JAVA_OPTS"}"
+JAVA_OPTS="${JAVA_OPTS:="$DEFAULT_JAVA_OPTS $MY_JAVA_OPTS}"}"
 
 start() {
     echo "Starting graylog-server ..."
