--- gitlab-runner-v11.7.0/Makefile.~1~	2019-01-22 12:06:07.000000000 +0000
+++ gitlab-runner-v11.7.0/Makefile	2019-02-15 06:39:16.704995377 +0000
@@ -14,9 +14,22 @@
 
 PACKAGE_CLOUD ?= ayufan/gitlab-ci-multi-runner
 PACKAGE_CLOUD_URL ?= https://packagecloud.io/
-BUILD_PLATFORMS ?= -os '!netbsd' -os '!openbsd'
 S3_UPLOAD_PATH ?= master
 
+ifeq ($(OS),Windows_NT)
+else
+	UNAME_S := $(shell uname -s)
+	ifeq ($(UNAME_S),SunOS)
+		export CGO_ENABLED ?= 1
+		BUILD_PLATFORMS ?= -os 'solaris'
+		DOCKER :=
+	else
+		export CGO_ENABLED ?= 0
+		BUILD_PLATFORMS ?= -os '!netbsd' -os '!openbsd'
+		DOCKER := docker
+	endif	
+endif
+
 # Keep in sync with docs/install/linux-repository.md
 DEB_PLATFORMS ?= debian/wheezy debian/jessie debian/stretch debian/buster \
     ubuntu/trusty ubuntu/xenial ubuntu/artful ubuntu/bionic \
@@ -52,8 +65,6 @@
               -X $(COMMON_PACKAGE_NAMESPACE).BRANCH=$(BRANCH) \
               -s -w
 GO_FILES ?= $(shell find . -name '*.go' | grep -v './.gopath/')
-export CGO_ENABLED ?= 0
-
 
 # Development Tools
 DEP = $(GOPATH_BIN)/dep
@@ -65,7 +76,7 @@
 
 .PHONY: clean version mocks
 
-all: deps docker build
+all: deps $(DOCKER) build
 
 include Makefile.docker.mk
 
